Storage/accessor server v0.9.0

boonwj opened the pull request:
Not v1.0.0 because blob storage is not done yet. 

This PR implements all the Accessor functionalities that interacts with the FHIR backend.

It is currently lacking sufficient tests, but that is currently in the works.

boonwj committed:
STORAGE: Initial commit for accessor V1 server

boonwj committed:
STORAGE: Add openapi go-gin-server code stubs

boonwj committed:
STORAGE: Add swaggerui static code files

boonwj committed:
STORAGE: Add swaggerui integration

- Remove servers from API yaml to disable unintended interactions from
ui

boonwj committed:
STORAGE: swaggerui to load spec from api folder

- Change openapi.yaml to load from /api folder via symbolic link
diff --git a/HoloStorageAccessor/cmd/swaggerui/openapi.yaml b/HoloStorageAccessor/cmd/swaggerui/openapi.yaml
deleted file mode 100644
index 4ca80b7..0000000
--- a/HoloStorageAccessor/cmd/swaggerui/openapi.yaml
+++ /dev/null
@@ -1,349 +0,0 @@
-openapi: 3.0.2
-info:
-  version: "0.1.4"
-  title: 'HoloStorage Accessor API'
-  description: 'API to access holograms and metadata from HoloStorage'
-
-# Added by API Auto Mocking Plugin
-servers:
-  - description: HoloStorage Sample Server URL
-    url: https://holoaccessor.com/api/0.1.4
-  - description: SwaggerHub API Auto Mocking
-    url: https://virtserver.swaggerhub.com/boonwj/HoloRepository/0.1.4
-
-paths:
-  /patients:
-    get:
-      summary: Mass query for patients metadata in HoloStorage
-      description: |-
-        Obtain patient information stored within HoloStorage.
-
-        Multiple IDs can be passed with the syntax `/patients?pid=id1,id2,id3,...,idx`
-      parameters:
-        - $ref: '#/components/parameters/PIDQuery'
-
-      responses:
-        200:
-          description: Successful pulling of patients metadata
-          content:
-            application/json:
-              schema:
-                $ref: '#/components/schemas/Patients'
-        400:
-          description: Unaccepted input
-
-  /patients/{pid}:
-    get:
-      summary: Get a single patient metadata in HoloStorage
-      description: Endpoint for a singular patient lookup
-      parameters:
-        - $ref: '#/components/parameters/PIDPath'
-      responses:
-        200:
-          description: Successful pulling of the patient's metadata
-          content:
-            application/json:
-              schema:
-                $ref: '#/components/schemas/Patient'
-        400:
-          description: Unaccepted input
-        404:
-          description: Patient cannot be found
-
-    put:
-      summary: Add or update basic patient information
-      description: |-
-        This endpoint allows you to add new patient information in HoloStorage. A new Patient will be created if PID does not exist, else it will be updated.
-
-        PID in path and request body will be checked as a means to ensure that the insertion is intended.
-      parameters:
-        - $ref: '#/components/parameters/PIDPath'
-      requestBody:
-        required: true
-        content:
-          application/json:
-            schema:
-              $ref: '#/components/schemas/PatientBasic'
-      responses:
-        200:
-          description: Successful modification
-        201:
-          description: Successful created
-        400:
-          description: Unaccepted input
-
-  /holograms:
-    get:
-      summary: Mass query for hologram metadata based on hologram ids
-      description: |-
-        Obtain hologram information stored within HoloStorage.
-
-        Multiple IDs can be passed with the syntax `/holograms?hid=id1,id2,id3,...,idx`
-      parameters:
-        - $ref: '#/components/parameters/HIDQuery'
-
-      responses:
-        200:
-          description: Successfully found all hologram metadata
-          content:
-            application/json:
-              schema:
-                $ref: '#/components/schemas/Holograms'
-        400:
-          description: Unaccepted input
-
-    post:
-      summary: 'Upload hologram to HoloStorage'
-      description: 'Hologram and associated metadata to be uploaded onto HoloStorage. The UUID will be generated automatically by the accessor'
-      requestBody:
-        required: true
-        content:
-          multipart/form-data:
-            schema:
-              $ref: '#/components/schemas/HologramUploadData'
-      responses:
-        200:
-          description: Success
-          content:
-            application/json:
-              schema:
-                type: object
-                properties:
-                  id:
-                    type: string
-                    description: The UUID of the uploaded hologram
-                    example: 32853d69-3e71-4c87-abf9-56f0eddefefc
-        400:
-          description: Unaccepted input
-
-  /holograms/{hid}:
-    get:
-      summary: Get a single hologram metadata based on the hologram id
-      description: Endpoint for a single hologram lookup
-      parameters:
-        - $ref: '#/components/parameters/HIDPath'
-      responses:
-        200:
-          description: Successfully found all hologram metadata
-          content:
-            application/json:
-              schema:
-                $ref: '#/components/schemas/Holograms'
-        400:
-          description: Unaccepted input
-        404:
-          description: Metadata of holograms were not found
-
-    delete:
-      summary: Delete a hologram in HoloStorage
-      description: Delete holograms based on ID
-      parameters:
-        - $ref: '#/components/parameters/HIDPath'
-      responses:
-        200:
-          description: Successful deletion of hologram
-        404:
-          description: Hologram cannot be located for deletion
-        500:
-          description: Error encountered during deletion process
-
-  /holograms/{hid}/download:
-    get:
-      summary: Download holograms models based on the hologram id
-      description: Initiates download operation for the specified hid
-      parameters:
-        - $ref: '#/components/parameters/HIDPath'
-      responses:
-        200:
-          description: Download a single hologram based on id
-          content:
-            model/gltf-binary:
-              schema:
-                type: string
-                format: binary
-                example: '[file content goes here]'
-        404:
-          description: Hologram is not found
-
-components:
-  parameters:
-    PIDQuery:
-      in: query
-      name: pid
-      required: true
-      schema:
-        type: string
-        format: uuid
-        example: 2321e0b8-99ea-4961-aab7-b65fbc8f9673
-      style: form
-      explode: false
-      description: The UUID assigned to the specific patient
-
-    PIDPath:
-      in: path
-      name: pid
-      required: true
-      schema:
-        type: string
-        format: uuid
-        example: 2321e0b8-99ea-4961-aab7-b65fbc8f9673
-      description: The UUID assigned to the specific patient
-
-    HIDQuery:
-      in: query
-      name: hid
-      required: true
-      schema:
-        type: string
-        format: uuid
-        example: abcdef12-3456-7890-ccbb-eeffaabbccdd
-      style: form
-      explode: false
-      description: The UUID assigned to a hologram
-
-    HIDPath:
-      in: path
-      name: hid
-      required: true
-      schema:
-        type: string
-        format: uuid
-        example: abcdef12-3456-7890-ccbb-eeffaabbccdd
-      description: The UUID assigned to a hologram
-
-  schemas:
-    PersonName:
-      description: Components that make up the name of a person
-      type: object
-      properties:
-        full:
-          type: string
-          example: Marvin Portebello
-        title:
-          type: string
-          example: Mr
-        given:
-          type: string
-          example: Marvin
-        family:
-          type: string
-          example: Portebello
-
-    Patients:
-      description: An array of patients metadata
-      type: array
-      items:
-        $ref: '#/components/schemas/Patient'
-
-    Patient:
-      title: Full Patient Info
-      description: Metadata of a single patient
-      allOf:
-        - $ref: '#/components/schemas/PatientBasic'
-        - type: object
-          properties:
-            holograms:
-              type: array
-              items:
-                $ref: '#/components/schemas/HologramBasic'
-
-    PatientBasic:
-      title: Basic Patient Info
-      description: The basic metadata of a patient
-      type: object
-      properties:
-        pid:
-          type: string
-          example: 2321e0b8-99ea-4961-aab7-b65fbc8f9673
-        name:
-          $ref: '#/components/schemas/PersonName'
-        gender:
-          type: string
-          example: male
-          enum: [male, female, unknown, other]
-        birthDate:
-          type: string
-          format: date
-
-    Holograms:
-      description: An array of hologram metadata
-      type: array
-      items:
-        $ref: '#/components/schemas/Hologram'
-
-
-    HologramUploadData:
-      title: Hologram Upload
-      description: Data structure to upload holograms to HoloStorage
-      allOf:
-        - $ref: '#/components/schemas/Hologram'
-        - type: object
-          properties:
-            hologramFile:
-              type: string
-              format: binary
-              example: [file content goes here]
-              description: The raw binary of the hologram model
-
-    Hologram:
-      title: Full Hologram Info
-      description: Metadata of a hologram
-      allOf:
-        - $ref: '#/components/schemas/HologramBasic'
-        - type: object
-          properties:
-            patient:
-              $ref: '#/components/schemas/PatientBasic'
-            author:
-              $ref: '#/components/schemas/Author'
-
-    HologramBasic:
-      title: Basic Hologram info
-      type: object
-      properties:
-        hid:
-          type: string
-          description: UUID of a hologram
-          example: 2321e0b8-99ea-4961-aab7-b65fbc8f9673
-        title:
-          type: string
-          description: Summary of the hologram
-          example: Malformed hip bones
-        description:
-          type: string
-          description: Further elaborations on the hologram with additional details
-          example: Malformed hip bones on an 8 year old child suffering from a birth defect
-        contentType:
-          type: string
-          description: The mimetype of the hologram file
-          example: model/gltf-binary
-        fileSizeInKb:
-          type: integer
-          description: The size of the hologram in KB
-          example: 5120
-        createdDate:
-          type: string
-          format: date-time
-          description: The date the hologram was generated
-          example: '2017-07-21T17:32:28Z'
-        creationMode:
-          type: string
-          enum: [GENERATE_FROM_IMAGING_STUDY, UPLOAD_EXISTING_MODEL]
-          description: The way the hologram was created
-          example: GENERATE_FROM_IMAGING_STUDY
-        pipelineId:
-          type: string
-          description: The id to the pipeline that was used in the generation of the hologram
-          example: bone-01
-
-    Author:
-      title: Author
-      description: Who and/or what authored the Hologram
-      type: object
-      properties:
-        aid:
-          type: string
-          description: The author ID
-          example: 2321e0b8-99ea-4961-aab7-b65fbc8f9673
-        name:
-          $ref: '#/components/schemas/PersonName'
diff --git a/HoloStorageAccessor/cmd/swaggerui/openapi.yaml b/HoloStorageAccessor/cmd/swaggerui/openapi.yaml
new file mode 120000
index 0000000..3d3d69b
--- /dev/null
+++ b/HoloStorageAccessor/cmd/swaggerui/openapi.yaml
@@ -0,0 +1 @@
+../../api/openapi.yaml
\ No newline at end of file

boonwj committed:
STORAGE: Add local relative package import

- Use local package import to enable ease of dockerfile building

boonwj committed:
STORAGE: Add util func to parse multiple queries

boonwj committed:
STORAGE: Add PersonName to HumanNameFHIR mapping

boonwj committed:
STORAGE: Add ToAPISpec() for HumanName

boonwj committed:
STORAGE: Convert Patient between FHIR and API spec

boonwj committed:
STORAGE: Add hologram api to fhir conversion

boonwj committed:
STORAGE: Add hologram FHIR to API conversion

boonwj committed:
STORAGE: Add Author FHIR and API spec converstion

boonwj committed:
STORAGE: Add Hologram schema from HologramUpload

HologramUpload schema is mostly similar to Hologram. Obtaining
information mapped to Hologram will be for FHIR DocRef operations.

boonwj committed:
STORAGE: Get DocRef data from HologramUpload model

This is the data that will be used to upload into FHIR datastore

boonwj committed:
STORAGE: Refactor formatting

Remove spacings and test names

boonwj committed:
STORAGE: Add method for URl construction

boonwj committed:
STORAGE: Fix author fhir format

Practitioner have an array of names in FHIR

boonwj committed:
STORAGE: Fix name field mapping in fhir patient

FHIR Patients have an array of names. Similar error to Practitioners

boonwj committed:
STORAGE: Add test for multiple FHIR HumanName

Affects Patients and Authors schemas

boonwj committed:
STORAGE: Make config to a package level var

boonwj committed:
STORAGE: Author GET endpoints

Add author GET functionalities for both for query and param endpoints.

boonwj committed:
STORAGE: Update author model to allow empty json

Struct components in authors changed to pointers, so that empty Authors
will result in a empty json output

boonwj committed:
STORAGE: Reimplement fhir query logic

- Removed overly tailored author query method
- Removed query related methods and structs from utils.go
- Added more generic fhir api methods that uses concurrency for queries

TODO: Reimplement author path endpoint

boonwj committed:
STORAGE: Add author path GET endpoint

boonwj committed:
STORAGE: add Author PUT endpoint

PUT endpoint for author implemented.

Reverted from pointer to personName value in author struct. JSON decoder
was unable to parse the results using the previous implementation.

But this results in name being created even when the struct is blank.

boonwj committed:
STORAGE: add patients GET endpoints

- Added patient GET endpoints.
- Noticed quite a bit of repeated code with authors. Possible refactor
area

boonwj committed:
STORAGE: add Patient PUT endpoint

Very similar to author's PUT endpoint.
Patients and Authors implementation can be revisited later

boonwj committed:
STORAGE: handler for unsupported fhir httpmethods

boonwj committed:
STORAGE: Refac hologram Id field to ID

Keeping it consistent with author and patients

boonwj committed:
STORAGE: Fix casing of default value for status

'current' is the acceptable word to be used instead of 'Current'

boonwj committed:
STORAGE: Refactor similar functions

Making author and patient get functions to be as similar as possible.

To be made into a more generic function later

boonwj committed:
STORAGE: Add GET endpoints for /holograms

GET functions are mostly similar to Author and Patients

boonwj committed:
STORAGE: Add responses for incomplete endpoints

boonwj committed:
STORAGE: Status code checks for missing resource

404 and 410 are returned by FHIR api server when resources are not
found.

To use those values as checks instead of comparing the captured ID.

boonwj committed:
STORAGE: Add hologram FHIR resource delete

TODO: when blob storage code is implemented this will require deletion
logic to be added here too

boonwj committed:
STORAGE: Fix non-empty json in author & patient

Change PersonName struct in OpenAPI Patient and Author type do
pointers. Change to pointers will allow blank results when json is
generated.

Change methods using PersonName to deal with it like a pointer

boonwj committed:
STORAGE: Fix non-empty json in holograms

Change Hologram and HologramUpload schemas to use pointer for time
objects.

This is similar to PersonName struct in Authors and Patients, the
time.Time type will be initialised when turned into JSON.

boonwj committed:
STORAGE: Refac mass query checks for missing data

Using error responses NOT_FOUND and GONE from FHIR server instead of
checking if the captured ID is different that what is expected

boonwj committed:
STORAGE: fix hologram author and subject reference

Author and Subject references was overlooked. Added referencing and
adding the prefix needed to allow searching in FHIR

boonwj committed:
STORAGE: Config change to .cfg for consistency

Other projects used .cfg for config. Updated it and README

boonwj committed:
STORAGE: Handle duplicates found in query

boonwj committed:
STORAGE: Parse hologram GET endpoint query

Parse queries fields in hologram endpoint. This will be used later to
craft different queries for holograms

boonwj committed:
STORAGE: Fix hologram mutli query GET returns

Result to be a map to an array of holograms instead of a single hologram

boonwj committed:
STORAGE: Allow query parameters for FHIRRequest

Used for subject and creationmode queries to FHIR's API

boonwj committed:
STORAGE: /holograms GET with pid and creationmode

pid and creationmode filtering uses the search functionality within
FHIR.

Results returned will be a FHIR Bundle instead of the resource itself.

This endpoint implementation expands all bundled and paginated results
from FHIR.

boonwj committed:
STORAGE: Fix multiple hologram endpoint results

The endpoint now correctly returns an empty array instead of an array
with an empty object.

boonwj committed:
STORAGE: Refac http calls in FHIRRequest to str

The strings up more explicitly in the code editor

boonwj committed:
STORAGE: POST endpoint for /hologram (fhir only)

Add post functionality /holograms to upload hologram metadata to the
backend FHIR server

boonwj committed:
STORAGE: Move swaggerui to third_party folder

This follows the convention provided by
https://github.com/golang-standards/project-layout

boonwj committed:
STORAGE: Restructure project layout

Project layout format based on
https://github.com/golang-standards/project-layout

boonwj committed:
STORAGE: Update dockerfile to build accessor

TODO: Current issue with how configurations are done. Might be a good
idea to switch to environment variables for config

boonwj committed:
STORAGE: Change config to environment variables

This would make it easier to work in docker

boonwj committed:
STORAGE: Update dockerfile with new configurations

- Add CA certificates to allow https connections
- Remove configuration file
- Add config.env file to be used for docker

boonwj committed:
Merge branch 'dev' into STORAGE/Accessor_Server_v1

Resolve README.md differences

boonwj committed:
STORAGE: Update README.md

- Installation of dependencies
- Requirements
- Docker build and run instructions
- Configuration fields

boonwj committed:
STORAGE: Update docker run instructions in README

Docker allows outgoing connections. So publishing incoming port is
sufficient.

nielsboecker commented on HoloStorageAccessor/README.md:
```diff
@@ -42,14 +55,29 @@ go test ./...
+## Configuration
+Accessor application uses the following environmental variables as configuration.
+
+If using docker, the environment configuration fields can be set via `config.env`.
```
Nice 👍

nielsboecker commented on HoloStorageAccessor/README.md:
````diff
@@ -42,14 +55,29 @@ go test ./...
 
 ```
-docker run --rm -it holostorageaccessor
+docker run -it --rm --env-file config.env -p 8080:8080 holo-storage-accessor
````
Add note how to verify it's up and running (`http://localhost:8080/app/1.0.0/ui/`, `curl "http://localhost:8080/api/1.0.0/patients?pid=foo,bar"`)

boonwj replied:
Note added to visit the index endpoint that has been changed to provide more useful information.

nielsboecker commented on HoloStorageAccessor/Dockerfile:
```diff
@@ -0,0 +1,20 @@
+FROM golang:1.12.7 AS build
+RUN mkdir /workspace
```
Could use a variable and be not in root dir, but this change is optional

`ARG WORKSPACE=/usr/src/workspace`

(note that this would have to be repeated in the second stage as ARG is scoped in such a way)

nielsboecker commented on HoloStorageAccessor/Dockerfile:
```diff
@@ -0,0 +1,20 @@
+COPY --from=build /workspace/api ./api
+COPY --from=build /workspace/holo-storage-accessor ./
+COPY --from=build /etc/ssl/certs/ca-certificates.crt /etc/ssl/certs/
+EXPOSE 8080/tcp
```
I usually don't specifiy the protocol, as using UDP would be quite exotic

nielsboecker commented on HoloStorageAccessor/README.md:
````diff
@@ -21,15 +21,28 @@ The following technologies are used in this component:
+To install program dependencies
+
+```
+go get -d -v ./...
````
I think it installed the dependencies to `HoloStorageAccessor/src/` - this should probably be gitignored then?

nielsboecker commented on HoloStorageAccessor/README.md:
```diff
@@ -21,15 +21,28 @@ The following technologies are used in this component:
 
 The API specification can be found in the `api/` directory. A deployed version of the interactive documentation is available [here](https://app.swaggerhub.com/apis/boonwj/HoloRepository/).
 
+When the application is deployed, the documentation can also be viewed at the `/app/1.0.0/ui` endpoint.
```
Make sure there are no confusions between `/app/` and `/api/`, as we discussed

boonwj replied:
Will be changed, both here and in router.go

nielsboecker commented on HoloStorageAccessor/cmd/holo-storage-accessor/main.go:
```diff
@@ -0,0 +1,23 @@
+import (
+	"log"
+
+	// sw "github.com/nbckr/HoloRepository-Core/HoloStorageAccessor/go"
```
Is this needed? Also, I assume `sw` is the nae defined by you, could it be something more readable?

boonwj replied:
Renamed import alias to "apiserver"

nielsboecker commented on HoloStorageAccessor/internal/openapi/api_default.go:
```diff
@@ -0,0 +1,389 @@
+ * API to access holograms and metadata from HoloStorage
+ *
+ * API version: 1.0.0
+ * Generated by: OpenAPI Generator (https://openapi-generator.tech)
```
Should not just say `Generated by` if it actually contains significant changes by you. First, you deserve the credit ;) secondly, when I read "generated" my reaction is "Don't touch, some tool will generate/overwrite this file"

boonwj replied:
Removed all instances of `generated by`. Mentions of openapi generator being used for server stub generation has already been added in the README

nielsboecker commented on HoloStorageAccessor/internal/openapi/fhir_api.go:
```diff
@@ -0,0 +1,141 @@
+		return FHIRResult{err: errors.New("Unsupported datatype")}
+	}
+
+	fmt.Println(fhirRequest)
```
Change to log or remove

boonwj replied:
wild dev printout removed

nielsboecker commented on HoloStorageAccessor/internal/openapi/model_hologram.go:
```diff
@@ -0,0 +1,171 @@
+	data := HologramMeta{}
+	err := json.Unmarshal([]byte(h.HologramMeta), &data)
+	if err != nil {
+		fmt.Println(err)
```
Change to log or remove

boonwj replied:
Changed to log

nielsboecker commented on HoloStorageAccessor/internal/openapi/model_hologram.go:
```diff
@@ -0,0 +1,171 @@
+	hologramData.CreationDescription = meta.CreationDescription
+
+	if len(h.Content) > 0 {
+		attachment_data := h.Content[0].Attachment
```
Use camelCase?

boonwj replied:
Changed this and another instance in utils.go.

nielsboecker commented on HoloStorageAccessor/internal/openapi/model_hologram_test.go:
```diff
@@ -0,0 +1,136 @@
+)
+
+func TestHologramToHologramDocumentReferenceFHIR(t *testing.T) {
+	ts_creation := time.Date(2019, 1, 2, 12, 30, 45, 0, time.UTC)
```
According to my IDE, camelCase would be preferred, but I am fine with ignoring this one

boonwj replied:
Yes I realised there are quite a number of snake cases in my tests code. Too much python :P I have ensured that the main code naming styles are in camel.

nielsboecker commented on HoloStorageAccessor/internal/openapi/routers.go:
```diff
@@ -0,0 +1,161 @@
+
+type AccessorConfig struct {
+	FhirURL       string `json:"fhirUrl"`
+	BlobStoreName string `json:"blobStoreName"`
```
Should it be called `BlobStorageName`? (and `BlobStorageKey`)

nielsboecker commented on HoloStorageAccessor/internal/openapi/routers.go:
```diff
@@ -0,0 +1,161 @@
+
+// Index is the index handler.
+func Index(c *gin.Context) {
+	c.String(http.StatusOK, "Hello World!")
```
Will this ever be served? If no, yes, maybe a more professional message like `"HoloStorageAccessor is running"`

nielsboecker commented on HoloStorageAccessor/internal/openapi/routers.go:
```diff
@@ -0,0 +1,161 @@
+	{
+		"Index",
+		http.MethodGet,
+		"/api/1.0.0/",
```
Maybe the prefix `"/api/1.0.0/"` (or even just the `1.0.0`) could be stored in a variable and added below with string interpolation (or just concatenation), so that a version change won't require many lines to change

boonwj replied:
Shifted prefix to a common variable

nielsboecker commented on HoloStorageAccessor/internal/openapi/utils.go:
```diff
@@ -0,0 +1,154 @@
+		case "creationDate":
+			creationDate, err := time.Parse(time.RFC3339, formData.Get(key))
+			if err != nil {
+				return HologramPostInput{}, errors.New(key + " is not in RFC3339 standards")
```
Probably `formData.Get(key)` will be the more helpful output (same for the similar lines below)

boonwj replied:
Changed both instances to return both the key and value in the error message

nielsboecker commented on HoloStorageAccessor/internal/openapi/utils.go:
```diff
@@ -0,0 +1,154 @@
+	var patientData Patient
+	var hologramData Hologram
+
+	for key, _ := range formData {
```
Would it make sense to actually use the values to, so that you don't have to `Get()` them individually when needed?

boonwj replied:
I just remembered that formData.Get(key) returns "" when the values are not set. The values by themselves are in an array and would be harder to parse.

nielsboecker reviewed (commented):
I'm not going to claim that I fully understand what is GOing on, but it does look really sound and elegant 🦄 . Just some aesthetic notes.

Btw, I see what you meant in terms of repetitions, but I think when you write a server application like this, it is really hard to avoid, and might not make the code  a lot shorter actually.

boonwj committed:
STORAGE: Refac README and Router

- UI path to /api/1.0.0/ui
- Refac basepath to read from a variable, for ease of changing
- Change welcome message at basic endpoint, to check for health
- README to note steps to check service status

boonwj committed:
STORAGE: Change url config checking during init

Fail fast and not when there's a request

boonwj committed:
STORAGE: Rename storage config variables

boonwj committed:
STORAGE: Return time value on parsing error

boonwj committed:
STORAGE: Update verification step in README

boonwj committed:
STORAGE: Updated error handling in fhir_api

- Removed unsupported datatype cases as the non-operation leads to weird
behaviour
- Removed wild println statement

boonwj committed:
STORAGE: Change wild fmt to log statement

boonwj committed:
STORAGE: Change snake variables to camel case

To keep main project naming consistent

boonwj committed:
STORAGE: Rename import alias used in main.go

boonwj committed:
STORAGE: Remove OpenAPI generation comment

Comments might lead to incorrect indiciation that code in the file were
generated by the OpenAPI generator.

nielsboecker reviewed (approved):
🕺  Fab!
